### PR TITLE
Apply "Parse fields in classes and generate correct class structs" in gir

### LINKED ISSUE #34: Expose Object structures so we can subclass them
To be able to subclass GObject, the exposed GObject structure need to have the right size, so also include the private fields. It is the case with basically any class but I hit the problem trying to [create a GObject subclass](https://github.com/thiblahute/rust-gobj-exp/blob/master/rust-gobject/src/main.rs)

Basically with that stupid patch it works:

``` diff
diff --git a/gobject-sys/src/lib.rs b/gobject-sys/src/lib.rs
index b3bc4e1..51f4af4 100644
--- a/gobject-sys/src/lib.rs
+++ b/gobject-sys/src/lib.rs
@@ -438,7 +438,12 @@ pub struct GBinding(c_void);
 #[repr(C)]
 pub struct GInitiallyUnowned(c_void);
 #[repr(C)]
-pub struct GObject(c_void);
+pub struct GObject {
+    pub g_type_instance: GTypeInstance,
+    ref_count: c_uint,
+    qdata: *mut glib::GData,
+
+}
```

I know everything is generated, but I can't build ( as I am getting ```rustc: /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/llvm/lib/IR/Instructions.cpp:1304: void llvm::StoreInst::AssertOK(): Assertion `getOperand(0)->getType() == cast<PointerType>(getOperand(1)->getType())->getElementType() && "Ptr must be a pointer to Val type!"' failed.``` while compiling `git2` ) thus I can't really play with it right now.

I will try to make some time to do it the right way but I would like to first have you opinion about it and maybe get some pointers about how that should happen.

### PR DESCRIPTION
**Pull request by EPashkin**
Example result of https://github.com/gtk-rs/gir/pull/276
WIP
cc @GuillaumeGomez
Closes #34

### CODE REVIEW

**Comment by EPashkin** [comment]
Related issue #34, #25.

**Comment by EPashkin** [comment]
Seems rust still have regression 😭

**Comment by GuillaumeGomez** [comment]
Wo, awesome! I'll just wait for appveyor and ignore beta/release for travis. This is really annoying. :-/

File: gdk-sys/src/lib.rs [hunk 1]
```diff
@@ -3712,39 +3712,58 @@ pub struct GdkWindowRedirect(c_void);
...
+    pub parent_instance: gobject::GObject,
+}
+
 #[repr(C)]
 pub struct GdkVisual(c_void);
+
 #[repr(C)]
 pub struct GdkWindow(c_void);
 
+
```
Comment (EPashkin) [inline gdk-sys/src/lib.rs]: Seems difference between last class and first interface

Comment (GuillaumeGomez) [inline gdk-sys/src/lib.rs]: Well, I don't really mind, just wanted to have more context. :)

Comment (EPashkin) [inline gdk-sys/src/lib.rs]: cc @eugene2k

**Comment by GuillaumeGomez** [comment]
appveyor failed with a "normal" error.

**Comment by EPashkin** [comment]
@GuillaumeGomez so you don't against this change?

**Comment by EPashkin** [comment]
Removed empty lines between classes and interfaces by removing first line in `generate_interfaces_structs`

**Comment by GuillaumeGomez** [comment]
I don't however, however, you must fix appveyor's failure first.

**Comment by EPashkin** [comment]
Fixed appveyor upgrading rust to 1.9 (stable will fail), so we need decide what we need ask @eugene2k to do.

**Comment by EPashkin** [comment]
Updated with results of https://github.com/gtk-rs/gir/pull/298

**Comment by EPashkin** [comment]
Updated, ready for merge

**Comment by GuillaumeGomez** [comment]
Thanks!

**Pull request merged**